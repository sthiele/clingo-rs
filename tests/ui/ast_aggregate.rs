use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = ast::Term::from(sym);
    let term2 = ast::Term::from(sym);
    let term3 = ast::Term::from(sym);
    let mut lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
    let lit2 = ast::Literal::from_term(ast::Sign::NoSign, &term2);
    let condition = vec![lit2];
    let cond = ast::ConditionalLiteral::new(&lit, &condition);
    let elements = vec![cond];
    let mut guard = ast::AggregateGuard::gt(term3);
    let agg = ast::Aggregate::new( &elements, Some(&guard), Some(&guard));
    
    guard =  ast::AggregateGuard::gt(term1);
    lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
    drop(elements);
    let _end = (guard, lit, agg);
}
