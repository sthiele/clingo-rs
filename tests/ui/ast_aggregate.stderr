error[E0506]: cannot assign to `guard` because it is borrowed
  --> $DIR/ast_aggregate.rs:16:5
   |
14 |     let agg = ast::Aggregate::new( &elements, Some(&guard), Some(&guard));
   |                                                    ------ borrow of `guard` occurs here
15 |     
16 |     guard =  ast::AggregateGuard::gt(term1);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `guard` occurs here
...
19 |     let _end = (guard, lit, agg);
   |                             --- borrow later used here

error[E0506]: cannot assign to `lit` because it is borrowed
  --> $DIR/ast_aggregate.rs:17:5
   |
11 |     let cond = ast::ConditionalLiteral::new(&lit, &condition);
   |                                             ---- borrow of `lit` occurs here
...
17 |     lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `lit` occurs here
18 |     drop(elements);
   |          -------- borrow later used here

error[E0505]: cannot move out of `elements` because it is borrowed
  --> $DIR/ast_aggregate.rs:18:10
   |
14 |     let agg = ast::Aggregate::new( &elements, Some(&guard), Some(&guard));
   |                                    --------- borrow of `elements` occurs here
...
18 |     drop(elements);
   |          ^^^^^^^^ move out of `elements` occurs here
19 |     let _end = (guard, lit, agg);
   |                             --- borrow later used here
